doc/statuscodes.md

```diff
@@ -0,0 +1,35 @@
+# Status codes and their use in gRPC
+
+gRPC uses a set of well defined status codes as part of the RPC API. All RPCs started at a client return  a `status` object composed of an integer `code` and a string `message`. The server-side can choose the status it returns for a given RPC.
+
+The gRPC client and server-side implementations may also generate and return `status` on their own when errors happen.  
+Only a subset of the pre-defined status codes are generated by the gRPC libraries. The following table lists these codes and summarizes the situations in which they are generated, either by the client or the server-side library implementation.
+
+| Case        | Code           | Generated at Client or server  |
+| ------------- |:-------------| :-----:|
+| Client Application cancelled the request	| GRPC_STATUS_CANCELLED | Both |
```

Comment:
> `GRPC_STATUS_CANCELLED`

Every language has a different way of prefixing, capitalizing, and joining words of the code name. So I would suggest removing the `GRPC_STATUS_` prefix here, for clarity, as it's redundant and doesn't buy us much in this context.